cli/cmd/plugin/ui/web/tanzu-ui/src/shared/components/DeployProgress/DeployTimeline/DeployTimeline.tsx

```diff
@@ -0,0 +1,139 @@
+// React imports
+import React, { useEffect, useState } from 'react';
+
+// Library imports
+import { CdsIcon } from '@cds/react/icon';
+import { circleIcon, ClarityIcons, dotCircleIcon, successStandardIcon } from '@cds/core/icon';
+
+// App imports
+import { StatusMessageData } from '../DeployProgress';
+import './DeployTimeline.scss';
+
+export const DeploymentStates = {
+    FAILED: 'failed',
+    RUNNING: 'running',
+    SUCCESSFUL: 'successful'
+};
+
+interface CurrentStatus {
+    msg: string;
+    status: string;
+    currentPhase: string;
+    finishedCount: number;
+    totalCount: number;
+}
+
+interface PropsData {
+    data?: StatusMessageData
+}
+
+ClarityIcons.addIcons(circleIcon, dotCircleIcon, successStandardIcon);
+
+function DeployTimeline(props:PropsData) {
+
+    const [phases, setPhases] = useState<Array<string>>([]);
+    const [currentPhaseIdx, setCurrentPhaseIdx] = useState<number>(0);
+    const [currentStatus, setCurrentStatus] = useState<CurrentStatus>({
+        msg: '',
+        status: '',
+        currentPhase: '',
+        finishedCount: 0,
+        totalCount: 0,
+    });
+
+    useEffect(()=>{
```

Comment:
if the data is from props, do we really need this useEffect?

Reply:
Good question, this is my first time passing props so I thought this might be the best solution. Every time there is a change to the props value, I need to call a method to parse the props. Is there a better way to call that method when props value changes?